src/storage/test/GetNeighborsBenchmark.cpp

```diff
@@ -0,0 +1,320 @@
+    nebula::storage::cpp2::GetNeighborsRequest req;
+    BENCHMARK_SUSPEND {
+        req = nebula::storage::buildRequest(vIds, playerProps, serveProps);
+        // where serve.startYear < value
+        auto* edgeProp = new std::string("startYear");
+        auto* alias = new std::string("101");
+        auto* edgeExp = new nebula::AliasPropertyExpression(new std::string(""), alias, edgeProp);
+        auto* priExp = new nebula::PrimaryExpression(value);
+        auto relExp = std::make_unique<nebula::RelationalExpression>(
+            edgeExp, nebula::RelationalExpression::Operator::LT, priExp);
+        req.set_filter(nebula::Expression::encode(relExp.get()));
+    }
+    for (decltype(iters) i = 0; i < iters; i++) {
+        auto* processor = nebula::storage::QueryBoundProcessor::instance(
+            gKV.get(), gSchemaMan.get(), nullptr, gExecutor.get());
+        auto f = processor->getFuture();
+        processor->process(req);
+        auto resp = std::move(f).get();
+        folly::doNotOptimizeAway(resp);
+    }
+}
+
+BENCHMARK(OneVertexOneProperty, iters) {
+    go(iters, {1}, {"name"}, {"teamName"});
+}
+BENCHMARK_RELATIVE(OneVertexOnePropertyWithFilter, iters) {
+    goFilter(iters, {1}, {"name"}, {"teamName"});
+}
+
+BENCHMARK(TenVertexOneProperty, iters) {
+    go(iters, {1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, {"name"}, {"teamName"});
+}
+BENCHMARK_RELATIVE(TenVertexOnePropertyWithFilter, iters) {
+    goFilter(iters, {1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, {"name"}, {"teamName"});
+}
+
+int main(int argc, char** argv) {
+    folly::init(&argc, &argv, true);
+    nebula::fs::TempDir rootPath("/tmp/QueryBoundBenchmarkTest.XXXXXX");
+    nebula::storage::setUp(rootPath.path());
+    folly::runBenchmarks();
+    gKV.reset();
+    return 0;
+}
+
+
+/*
+40 processors, Intel(R) Xeon(R) CPU E5-2690 v2 @ 3.00GHz.
+release
+
+--max_rank=1000 --filter_ratio=0.1
+============================================================================
+GetNeighborsBenchmark.cpprelative                         time/iter  iters/s
+============================================================================
+OneVertexOneProperty                                       624.38us    1.60K
+OneVertexOnePropertyWithFilter                    88.79%   703.23us    1.42K
+TenVertexOneProperty                                         6.00ms   166.76
+TenVertexOnePropertyWithFilter                    89.82%     6.68ms   149.78
+============================================================================
+
+--max_rank=1000 --filter_ratio=0.5
+============================================================================
+GetNeighborsBenchmark.cpprelative                         time/iter  iters/s
+============================================================================
+OneVertexOneProperty                                       629.54us    1.59K
+OneVertexOnePropertyWithFilter                    72.33%   870.39us    1.15K
```

Comment:
It seems the more data filtered out, the better performance acquired.   
So the bottleneck is encoder(RowWriter) ?

Reply:
Hard to say whether bottleneck is encoder or not. With bigger `filter_ratio`, it will need to eval the exp which is more likely to be useless. But with a small `filter_ratio`, we can filter out more data, so there will be less data to encode. We can only say with a bit `filter_ratio`, it has worse performance.

The result in 2.0 is same.

Reply:
What if we comment the collector?  What's the performance looks like?

Reply:
Looks like this, do not collect any value.
![image](https://user-images.githubusercontent.com/13706157/89407171-e3f41080-d750-11ea-8694-57971ca38726.png)

Reply:
Good job.
By the way, I think that even with the same data, there is no comparison between filters and no filters, right?

Reply:
Yeah, you're right.

Reply:
> Looks like this, do not collect any value.
> ![image](https://user-images.githubusercontent.com/13706157/89407171-e3f41080-d750-11ea-8694-57971ca38726.png)

Em.... So we can ensure the cost comes from encoder.

Reply:
Let's run the benchmark for a long time,  and check the perf top about it.  @critical27